### PR TITLE
Start architecture docs

### PR DESCRIPTION
**Pull request by seb-cr**
As discussed on Thursday, I've added a new page where we can put high-level docs about architecture that spans multiple services. To start things off I've recreated the supporter data flow diagram using mermaidjs and added a gitbook plugin to render it.

### CODE REVIEW

**Review by dipasqualew — APPROVED** [review]

**Comment by dipasqualew** [comment]
Looks great, and it's also great we can see the preview:
https://deploy-preview-17--comicrelief-engineering-handbook.netlify.app/docs/architecture.html

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Architecture
+***
+
+While we prefer in-code and live documentation over wiki pages, these are best suited to single services and we don't have a place to document broader processes that involve the interaction of several services. This section aims to provide that overview of how various services fit together.
+
+## Supporter data flow
+
+```mermaid
+graph LR
+  fundrasing-signups["Fundraising Sign-ups (Tom)"] --- event
```
Comment (tomisaacmarshall) [inline docs/architecture.md]: Most minor thing of all time but there's a missing `i` here and also, would be nice to stick to the same naming conventions - I've been going with 'Fundraiser Signups'

Comment (seb-cr) [reply docs/architecture.md]: Thanks, well spotted! Agree that names should be consistent everywhere.

**Comment by tomisaacmarshall** [comment]
Looking good!

Wondering it it's possible to flesh it out a bit more though? 

I am a bit behind with all this, think I was only in some of the meetings and was somewhat distracted at the time by other things, so would be nice to have a bit more info. E.g. a brief summary of why we decided to create this 'supporter-event-service', what problem it's solving etc, and perhaps what the different roles of the user and erp services are?

Or do we think that stuff should just go in the repos themselves? If so - maybe the boxes for the different services can include links to their readmes?

Also wondering if the front-facing services are going to be accessing data from the user/erp services directly, and if that needs mapping out if so? It seems to just cover ingesting data at the mo

**Comment by tomisaacmarshall** [comment]
> Looking good!
> 
> Wondering it it's possible to flesh it out a bit more though?
> 
> I am a bit behind with all this, think I was only in some of the meetings and was somewhat distracted at the time by other things, so would be nice to have a bit more info. E.g. a brief summary of why we decided to create this 'supporter-event-service', what problem it's solving etc, and perhaps what the different roles or the user and erp services are?
> 
> Or do we think that stuff should just go in the repos themselves? If so - maybe the boxes for the different services can include links to their readmes?
> 
> Also wondering if the front-facing services are going to be accessing data from the user/erp services directly, and if that needs mapping out if so? It seems to just cover ingesting data at the mo

Sorry just read the description again where you are very clear that this is just a start! 👍

**Review by tomisaacmarshall — APPROVED** [review]

**Comment by seb-cr** [comment]
Yeah, this is really just intended to make a place where this documentation can exist – I don't currently feel that my knowledge is good enough yet to expand on it, so am hoping that other people will add to this page once it's merged. I love the idea of links to each service in the diagram though so will try to do that now!

As for what should go where... I'm inclined to say that the "why" and the "what" should go in these architecture docs, and the "how" should go in the repo. But the repo should also state what it is, so there is risk of duplicating explanations between the high-level docs and each repo's docs, which we ideally want to minimise.

Perhaps one approach could be to write the "why"/"what" in such a way that we can use the same text in both places. Or just extensively hyperlink between them, e.g. starting each service's readme with "Please refer to the Architecture Docs for an explanation of what this service is and why it is needed". But we can explore different options and see what works :)

**Comment by corinja** [comment]
Just a minor thing, and this is an observation not a change request, but the labels on the graph aren't wide enough for the text.

**Review by corinja — APPROVED** [review]

**Comment by dipasqualew** [comment]
@tomisaacmarshall I think services will need to access data from both User Service and ERP Service, for example user logins and user accessing their historic transactions. However, every time we query User Service and ERP Service, it will need to be for non-critical paths. For all critical paths we need to assume both services will be down and should not be relied upon.

Had a chat with @adamclark-dev and @lebaz20 about critical data. Supporter Events Service gives us assurance writes will be successful (eventually). If critical information is required by the service (for example, the email address of a user), then it should be stored in the service itself

**Comment by tomisaacmarshall** [comment]
> Yeah, this is really just intended to make a place where this documentation can exist – I don't currently feel that my knowledge is good enough yet to expand on it, so am hoping that other people will add to this page once it's merged. I love the idea of links to each service in the diagram though so will try to do that now!
> 
> As for what should go where... I'm inclined to say that the "why" and the "what" should go in these architecture docs, and the "how" should go in the repo. But the repo should also state what it is, so there is risk of duplicating explanations between the high-level docs and each repo's docs, which we ideally want to minimise.
> 
> Perhaps one approach could be to write the "why"/"what" in such a way that we can use the same text in both places. Or just extensively hyperlink between them, e.g. starting each service's readme with "Please refer to the Architecture Docs for an explanation of what this service is and why it is needed". But we can explore different options and see what works :)

Cool! Yeah cross referencing between the services and this handbook is probably a good idea

**Comment by seb-cr** [comment]
> Just a minor thing, and this is an observation not a change request, but the labels on the graph aren't wide enough for the text.

@corinja Seems to be a random glitch with mermaid that goes away if you reload the page.

**Comment by tomisaacmarshall** [comment]
> @tomisaacmarshall I think services will need to access data from both User Service and ERP Service, for example user logins and user accessing their historic transactions. However, every time we query User Service and ERP Service, it will need to be for non-critical paths. For all critical paths we need to assume both services will be down and should not be relied upon.
> 
> Had a chat with @adamclark-dev and @lebaz20 about critical data. Supporter Events Service gives us assurance writes will be successful (eventually). If critical information is required by the service (for example, the email address of a user), then it should be stored in the service itself

Cheers! Just the sort of stuff that would be nice to add to these docs :)

**Pull request merged**